scheduler/test/cook/test/scheduler/constraints.clj

```diff
@@ -141,6 +146,16 @@
                          (getCurrAvailableResources [_] (sched/->VirtualMachineLeaseAdapter k8s-gpu-offer 0)))
                        nil))
           (str "GPU task on GPU host with the correct number of GPUs should succeed"))
+      (is (not (.isSuccessful
+                 (.evaluate (constraints/fenzoize-job-constraint (constraints/build-gpu-host-constraint gpu-job-4))
+                            (sched/make-task-request db gpu-job-4 nil)
+                            (reify com.netflix.fenzo.VirtualMachineCurrentState
+                              (getHostname [_] "test-host")
+                              (getRunningTasks [_] [])
+                              (getTasksCurrentlyAssigned [_] [gpu-task-assignment])
+                              (getCurrAvailableResources [_] (sched/->VirtualMachineLeaseAdapter k8s-gpu-offer 0)))
+                            nil)))
+          (str "GPU task on GPU host with the correct number of GPUs but already has task assigned should fail"))
```

Comment:
What about "We only allow one GPU job per VM. This VM already has a GPU job, so shouldn't get another"?

Reply:
thanks that makes the comment way less confusing, updated to something pretty similar.